src/lib/parser.c

```diff
@@ -1620,7 +1641,7 @@ RdbStatus elementEndOfFile(RdbParser *p) {
         }
     }
 
-    CALL_COMMON_HANDLERS_CB(p, handleEndRdb);
+    CALL_COMMON_HANDLERS_CB_NO_ARGS(p, handleEndRdb);
```

Comment:
so the transition to c99 was easy? i'm curious what else did it involve?
nanosleep and static are related?

Reply:
In addition I had few functions that are not available on c99,
which are nanosleep(), kill(), strdup(), setenv().

At two files I added at the begining the following 
```
# test_common.c
/* feature-test-macros POSIX.1-2008 for: kill(), strdup(), setenv() */
#define _POSIX_C_SOURCE 200809L

# respToRedisLoader.c
/* feature-test-macros POSIX.1-2008 for: nanosleep() */
#define _POSIX_C_SOURCE 200809L
```
If it will bother us in the future we can try eliminate them as well.

Reply:
In addition I had few functions that are not available on c99,
which are nanosleep(), kill(), strdup(), setenv().

At two files I added at the begining the following 
```
# test_common.c
/* feature-test-macros POSIX.1-2008 for: kill(), strdup(), setenv() */
#define _POSIX_C_SOURCE 200809L

# respToRedisLoader.c
/* feature-test-macros POSIX.1-2008 for: nanosleep() */
#define _POSIX_C_SOURCE 200809L
```
If it will bother us in the future we can try eliminate them as well.

Reply:
that's perfectly fine.

Reply:
@tezc , to workaround it I refactored the *HANDLERS_CB macros to support either with or without arguments.
If you need the very same macros, then do your own temporary change to workaround it and consume this PR once pushed.

IMO, our best effort is to stick to the standard (c99) and if we cannot, then see what are the options to allow it gracefully.

Reply:
@moticless I'm not in hurry, I can definitely wait your PR :) IIUC you will push that change to this PR? 

Btw, there are a few more places like `ouput_fprintf(...)`, `UNUSED(...)`. You can change CC = clang and check it if you want.